src/main/java/bob/task/TaskList.java

```diff
@@ -0,0 +1,332 @@
+     * @param itemIndex Index of the item in the TaskList.
+     * @return True if index is an index within TaskList, else false.
+     */
+    public boolean isValidRecord(int itemIndex) {
+        return (itemIndex > 0 && itemIndex <= records.size());
+    }
+
+
+
+    /**
+     * Removes the task from records, based on the task's index in records.
+     * @param input Input given by the user.
+     */
+    public void delete(String input) {
+        try {
+            String[] separateKeyword = input.split(" ", 2); //separate the keyword from the rest of string
+            if (separateKeyword.length == 1) {
+                throw new InvalidTaskException("OOPS!!! The description of delete cannot be empty.");
+            }
+            if (!isValidRecord(Integer.parseInt(separateKeyword[1]))) {
+                throw new InvalidTaskException("Invalid input. Integer required between range of record items.");
+            }
+
+            Task taskToDelete = records.get(Integer.parseInt(separateKeyword[1]) - 1);
+            String immediateAdd = "Noted. I've removed this task:\n\t"
+                    + taskToDelete.getTaskListItem()
+                    + "\n\t"
+                    + "Now you have "
+                    + (String.valueOf(latestRecordedIndex - 1))
+                    + " tasks in the list.";
+            Ui.printLines(immediateAdd);
+            this.records.remove(Integer.parseInt((separateKeyword[1]).trim()) - 1);
+            this.latestRecordedIndex -= 1;
+        } catch (InvalidTaskException e) {
+            System.err.println((e.getMessage()));
+        }
+    }
+
+    /**
+     * Updates whether the task in the record is completed or not completed.
+     * @param inputWords String array of the words given as input.
+     * @param isCompleted Whether the task is marked as completed or incompleted.
+     */
+    public void updateMark(String input, String[] inputWords, boolean isCompleted) {
+        if (inputWords.length == 1) {
+            System.out.println("Please input which item number you want to mark.");
+        } else if (this.records.size() < Integer.valueOf(inputWords[1]) || Integer.valueOf(inputWords[1]) <= 0) {
+            System.out.println("Item index out of range.");
+        } else {
+            Task currTask = this.records.get(Integer.parseInt(inputWords[1]) - 1);
+            if (isCompleted) {
+                currTask.markTask(true);
+            } else {
+                currTask.markTask(false);
+            }
+        }
+    }
+
+    /**
+     * Adds a task to records.
+     * @param input Input given by a user.
+     * @param inputWords Partially processed array of input.
+     */
+    public void addTask(String input, String[] inputWords) {
+        try {
+            String keyword = inputWords[0];
+            Task newTask = getTask(keyword, input); //initialise the exact Task class
+            String immediateAdd = "Got it. I've added this task:\n\t"
+                    + "  ["
```

Comment:
Good that you always remember when to leave white spaces especially for operators.